src/Features/Core/Portable/Completion/CompletionHelper.cs

```diff
@@ -234,8 +234,15 @@ private int CompareMatches(PatternMatch match1, PatternMatch match2, CompletionI
             // and 'False' (with the latter being 'Preselected').  Both will be a prefix match.
             // And because we are ignoring case, neither will be seen as better.  Now, because
             // 'False' is preselected we pick it even though 'foo' matches 'f' case sensitively.
+
+            // At this point, we've only done case insesitive comparisons. Consider:
+            // using System.IO; ...
+            // string path = Path$$ 
+            // System.IO.Path is a better case-sensitive match but has a worse
+            // preselection score from target typing
```

Comment:
It seems like the code below:

```
// Now compare the matches again in a case sensitive manner.  If everything was
// equal up to this point, we prefer the item that better matches based on case.
diff = match1.CompareTo(match2, ignoreCase: false);
```

Needs to be better integrated with this logic.  i.e. target typing should happen *after* this check.